modules/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -0,0 +1,293 @@
+        }
+    }
+    return contentType;
+}
+
+@Description {value:"Represent MIME specific base64 encoder. This follows RFC 2045 for encoding operation."}
+public struct MimeBase64Encoder {
+}
+
+@Description {value:"Represent MIME specific base64 decoder. This follows RFC 2045 for decoding operation."}
+public struct MimeBase64Decoder {
+}
+
+@Description {value:"Represent quoated-printable encoder."}
+public struct QuotedPrintableEncoder {
+}
+
+@Description {value:"Represent quoated-printable decoder."}
+public struct QuotedPrintableDecoder {
+}
+
+@Description {value:"Encode byte array using MIME Base64 encoding scheme."}
+@Param {value:"encoder: Represent MIME specific base64 encoder"}
+@Param {value:"content: the byte array to encode"}
+@Return {value:"return resulting encoded bytes"}
+public native function <MimeBase64Encoder encoder> encode (blob content) (blob);
+
+@Description {value:"Encode a given string using MIME Base64 encoding scheme. First the given string will be
+converted to a byte array with the given charset encoding. If the charset given is null default 'UTF-8' will be used.
+ Then that byte array will be encoded using MIME Base64 encoding scheme and a new string will be constructed with the
+  given charset."}
+@Param {value:"encoder: Represent MIME specific base64 encoder"}
+@Param {value:"content: string to encode"}
+@Param {value:"charset: charset used in the given string and the resulting string"}
+@Return {value:"return resulting encoded string"}
+public native function <MimeBase64Encoder encoder> encodeString (string content, string charset) (string);
+
+@Description {value:"Decode byte array using MIME Base64 encoding scheme."}
+@Param {value:"encoder: Represent MIME specific base64 decoder"}
+@Param {value:"content: the byte array to decode"}
+@Return {value:"return resulting decoded bytes"}
+public native function <MimeBase64Decoder decoder> decode (blob content) (blob);
+
+@Description {value:"Decode a given string using MIME Base64 decoding scheme. First the given string will be
+converted to a byte array with the given charset encoding. If the charset given is null default 'UTF-8' will be used.
+ Then that byte array will be decoded using MIME Base64 decoding scheme and a new string will be constructed with the
+  given charset."}
+@Param {value:"encoder: Represent MIME specific base64 decoder"}
+@Param {value:"content: string to decode"}
+@Param {value:"charset: charset used in the given string and the resulting string"}
+@Return {value:"return resulting decoded string"}
+public native function <MimeBase64Decoder decoder> decodeString (string content, string charset) (string);
+
+@Description {value:"Utility function used internally to get the bytes from a given channel."}
+@Param {value:"channel: A ByteChannel"}
+@Return {value:"return a byte array"}
+function readAll (io:ByteChannel channel) (blob) {
+    blob bytes;
+    int numberOfBytesRead;
+    bytes, numberOfBytesRead = channel.readAllBytes();
```

Comment:
numberOfBytesRead is not used. So can we simplify this to `var bytes, _ = channel.readAllBytes();
return bytes;
`